moirae/estimators/online/filters/conversions.py

```diff
@@ -0,0 +1,191 @@
+    :class:`~moirae.estimators.online.filters.distributions.MultivariateRandomDistribution` to tell it how it should be
+    transformed. It is used at the :class:`~moirae.estimators.online.filters.base.ModelWrapper` to help translate the
+    output from :class:`~moirae.estimators.online.filters.base.BaseFilter` to numerical values that are pertinent to
+    :class:`~moirae.models.base.CellModel`.
+    """
+
+    @abstractmethod
+    def transform_samples(self, samples: np.ndarray) -> np.ndarray:
+        """
+        Transforms a set of individual points, considered to be independent samples of the same
+        :class:`~moirae.estimators.online.filters.distributions.MultivariateRandomDistribution`.
+
+        Args:
+            samples: np.ndarray of samples to be transformed. Its shape must be (num_samples, num_coordinates) or
+                (num_coordinates,)
+
+        Returns:
+            transformed_samples: np.ndarray of the transformed points
+        """
+        raise NotImplementedError('Implement in child class!')
+
+    @abstractmethod
+    def transform_covariance(self, covariance: np.ndarray) -> np.ndarray:
+        """
+        Transforms the covariance of a
+        :class:`~moirae.estimators.online.filters.distributions.MultivariateRandomDistribution`.
+
+        Args:
+            covariance: 2D np.ndarray corresponding to the covariance to be transformed
+
+        Returns:
+            transformed_covariance: transformed covariance matrix
+        """
+        raise NotImplementedError('Implement in child class!')
+
+    @abstractmethod
+    def inverse_transform_samples(self, transformed_samples: np.ndarray) -> np.ndarray:
+        """
+        Performs the inverse tranformation of that given by
+        :meth:`~moirae.estimators.online.filters.transformations.BaseTransform.transform_points`.
+
+        Args:
+            transformed_samples: np.ndarray of points to be inverse transformed
+
+        Returns:
+            samples: np.ndarray corresponding to re-converted points
+        """
+        raise NotImplementedError('Implement in child class!')
+
+    @abstractmethod
+    def inverse_transform_covariance(self, transformed_covariance: np.ndarray) -> np.ndarray:
+        """
+        Performs the inverse tranformation of that given by
+        :meth:`~moirae.estimators.online.filters.transformations.BaseTransform.transform_covariance`.
+
+        Args:
+            transformed_covariance: np.ndarray of transformed covariance to be converted back
+
+        Returns:
+            covariance: np.ndarray corresponding to re-converted covariance
+        """
+        raise NotImplementedError('Implement in child class!')
+
+
+class IdentityConversionOperator(ConversionOperator):
+    """
+    Class that implements simple identity operation, that is, it does not change the inputs.
+    """
+    def transform_samples(self, samples: Optional[np.ndarray]) -> np.ndarray:
+        if samples is None:
```

Comment:
The base class's definition says samples is not optional, so let's not change it in the subclass. Also, why would the only argument to a function be optional?

Reply:
I thought I had a reason, but I cannot remember it now, so it's probably not important. Based on good code design, if the reason is important, we will have to find a different work around later, because, as you said, this should not be optional. Fixing it.